src/main/java/pl/piekoszek/gorskimatches/challange/ChallengeController.java

```diff
@@ -23,51 +23,36 @@ UUID createChallenge() {
     }
 
     @PostMapping("score")
-    void saveRegisteredUserAndGetResults(@Email(required = false) String email, @RequestBody ChallengeResult challengeResult) throws MessagingException {
+    void saveUsersScore(@Email(required = false) String email, @RequestBody ChallengeResult challengeResult) throws MessagingException {
         if (email != null) {
-            challengeService.saveRegisteredUserResult(challengeResult, email);
+            challengeService.saveUserWithEmailAndEmailWithResult(challengeResult, email);
             return;
         }
-        challengeService.saveNonRegisteredUserResult(challengeResult);
-        challengeService.resultForRegisteredUser(challengeResult);
+        challengeService.saveUserWithoutEmailAndEmailWithResult(challengeResult);
     }
 
     @GetMapping("resultForNonRegistered/{uuid}")
```

Comment:
Existance of this endpoint looks wired to me, wouldn't be nicer to save who won in `Challenge`? and then instead of this endpoint, use 
```
@GetMapping("{uuid}")
Challenge getChallenge(@PathVariable("uuid") UUID uuid) {
```

having who won could potentially simplify front-end, but I'm not sure